### PR TITLE
drivers: gpio: Avoid conflicts

### PR DESCRIPTION
**Pull request by rettichschnidi**
By sorting the lines alphabetically, conflicts can be reduced.

## Background

For a while now, I am working on support for a new SoC. Often, when rebasing my WIP branch on the most recent main, conflicts happen because everyone adds new drivers at the bottom. By biting the bullet now, this kind of conflict can (most of the time) be avoided in the future.

### CODE REVIEW

File: drivers/gpio/CMakeLists.txt [hunk 1]
```diff
@@ -4,95 +4,97 @@ zephyr_syscall_header(${ZEPHYR_BASE}/include/zephyr/drivers/gpio.h)
 
 zephyr_library()
 
+# Please keep this list sorted alphabetically
```
Comment (henrikbrixandersen) [inline drivers/gpio/CMakeLists.txt:7]:
```suggestion
# zephyr-keep-sorted-start
```

File: drivers/gpio/CMakeLists.txt [hunk 2]
```diff
@@ -4,95 +4,97 @@ zephyr_syscall_header(${ZEPHYR_BASE}/include/zephyr/drivers/gpio.h)
...
-zephyr_library_sources_ifdef(CONFIG_GPIO_TLE9104    gpio_tle9104.c)
-zephyr_library_sources_ifdef(CONFIG_GPIO_ALTERA_PIO gpio_altera_pio.c)
-zephyr_library_sources_ifdef(CONFIG_GPIO_BCM2711    gpio_bcm2711.c)
-zephyr_library_sources_ifdef(CONFIG_GPIO_RENESAS_RA gpio_renesas_ra.c)
-zephyr_library_sources_ifdef(CONFIG_GPIO_ENE_KB1200 gpio_ene_kb1200.c)
-zephyr_library_sources_ifdef(CONFIG_GPIO_RZT2M    gpio_rzt2m.c)
-zephyr_library_sources_ifdef(CONFIG_GPIO_AMBIQ      gpio_ambiq.c)
-zephyr_library_sources_ifdef(CONFIG_GPIO_BRCMSTB    gpio_brcmstb.c)
+zephyr_library_sources_ifdef(CONFIG_GPIO_SHELL      gpio_shell.c)
+zephyr_library_sources_ifdef(CONFIG_USERSPACE   gpio_handlers.c)
```
Comment (henrikbrixandersen) [inline drivers/gpio/CMakeLists.txt:97]:
```suggestion
zephyr_library_sources_ifdef(CONFIG_USERSPACE   gpio_handlers.c)
# zephyr-keep-sorted-stop
```

File: drivers/gpio/Kconfig [hunk 3]
```diff
@@ -86,162 +86,85 @@ config GPIO_ENABLE_DISABLE_INTERRUPT
 	  on/off the interrupt signal without changing other registers, such as
 	  pending register, etc. The driver must implement it to work.
 
-
+# Please keep this list sorted alphabetically
```
Comment (henrikbrixandersen) [inline drivers/gpio/Kconfig:89]:
```suggestion
# zephyr-keep-sorted-start
```

File: drivers/gpio/Kconfig [hunk 4]
```diff
@@ -86,162 +86,85 @@ config GPIO_ENABLE_DISABLE_INTERRUPT
...
-
-source "drivers/gpio/Kconfig.rzt2m"
-
-source "drivers/gpio/Kconfig.ambiq"
-
-source "drivers/gpio/Kconfig.brcmstb"
+source "drivers/gpio/Kconfig.xec"
+source "drivers/gpio/Kconfig.xlnx"
+source "drivers/gpio/Kconfig.xlnx_ps"
+source "drivers/gpio/Kconfig.xmc4xxx"
```
Comment (henrikbrixandersen) [inline drivers/gpio/Kconfig:168]:
```suggestion
source "drivers/gpio/Kconfig.xmc4xxx"
# zephyr-keep-sorted-stop
```

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
Please use the markers for ensuring future sorting is checked by CI.

**Review by henrikbrixandersen — APPROVED** [review]

**Review by pdgendt — APPROVED** [review]

**Pull request merged**